### PR TITLE
FFI and Rust version updates

### PR DESCRIPTION
**Pull request by andrewwhitehead**
- Use i64 consistently instead of usize in FFI (to avoid platform differences, and because it's supposed to be easier to deal with in Java)
- Add C header file
- Update Go version in wrapper
- Update to Rust 2021 edition

### CODE REVIEW

**Comment by andrewwhitehead** [comment]
@blu3beri This would probably have some impact on the React wrapper.

**Comment by blu3beri** [comment]
Thanks for the mention. I will modify the wrapper accordingly. I am working on compiling the library for Android right now and zmq is giving some issues. (I don't think the vendored feature works for Android, but I will look more into it next week).

File: libindy_vdr/include/cbindgen.toml
Comment (blu3beri) [inline libindy_vdr/include/cbindgen.toml:140]: I might be mistaken here as I never really used cbindgen with more than the default config, but if this is uncommented and `parse_deps = true` then it should generate the ByteBuffer correctly, right?

**Comment by blu3beri** [comment]
Is there any eta on the merging of this PR? If a reviewer is needed I can take a complete look.

**Comment by swcurran** [comment]
Please do @blu3beri . Thanks!

File: libindy_vdr/include/libindy_vdr.h
Comment (blu3beri) [inline libindy_vdr/include/libindy_vdr.h:45]: Regenerating the cbindgen seems to remove this diff:

```diff
@@ -40,14 +40,6 @@ typedef int64_t ErrorCode;
 #endif // __cplusplus

 typedef int64_t CallbackId;
-typedef int64_t PoolHandle;
-typedef int64_t RequestHandle;
-typedef const char *FfiStr;
-
-typedef struct {
-    int64_t len;
-    uint8_t *data; // note: nullable
-} ByteBuffer;

 #ifdef __cplusplus
 extern "C" {
```

adding this to `cbindgen.toml` will add `PoolHandle` and `RequestHandle` to the generation. 

```toml
after_includes = """
typedef int64_t PoolHandle;
typedef int64_t RequestHandle;
"""
```

and adding this would include the ByteBuffer and FfiStr:

```toml
[parse]
parse_deps = true
include = ["ffi-support"]
```

File: libindy_vdr/src/common/handle.rs
Comment (blu3beri) [inline libindy_vdr/src/common/handle.rs:10]: I am not too sure how, but can we expose the handle type so we do not have to add it manually in the header? so the `$newtype` would expose it to the header as `typedef int64_t $newtype;`

**Review by blu3beri — APPROVED** [review]
I think my "approve" does anything functional.

**Comment by swcurran** [comment]
It looks like you want @andrewwhitehead to make changes before we merge, right?  Once that's settled/done, I'll do an approval.

Are you an appropriate person and would you like to be a maintainer for this repo? Love to have help on this.

**Comment by blu3beri** [comment]
> It looks like you want @andrewwhitehead to make changes before we merge, right? Once that's settled/done, I'll do an approval.

Yes, I did `approve` as it was not anything major and more some advice for codegen. 
 
> Are you an appropriate person and would you like to be a maintainer for this repo? Love to have help on this.

I can definitely help with pull requests, but my knowledge of the internal is quite limited.

File: libindy_vdr/include/libindy_vdr.h
Comment (andrewwhitehead) [reply libindy_vdr/include/libindy_vdr.h:87]: Thanks, yes the parsing of the `ffi_support` crate wasn't working at all for me before, but it is now, so go figure. I adjusted the code generation for the handle types so manual additions aren't required anymore.

**Pull request merged**
